Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp

```diff
@@ -113,36 +119,118 @@ namespace AzToolsFramework
+    int DPEModelNode::ColumnOfChild(DPEModelNode* const childNode) const
     {
-        auto& jsonBackend = GetModel()->GetBackend();
-        for (size_t arrayIndex = 0, numIndices = domVal.ArraySize(); arrayIndex < numIndices; ++arrayIndex)
+        return m_columnChildren.indexOf(childNode);
+    }
+
+    void DPEModelNode::SetValue(const AZ::Dom::Value& domVal, bool notifyView)
+    {
+        AZStd::string domName = domVal.GetNodeName().GetCStr();
+
+        // these will only be used if we're notifying the view
+        DPEDebugModel* theModel = nullptr;
+        QPersistentModelIndex myIndex;
+        if (notifyView)
         {
-            auto& currentValue = domVal[arrayIndex];
-            AZStd::string currentName = currentValue.GetNodeName().GetCStr();
+            theModel = GetModel();
+            myIndex = theModel->GetIndexFromNode(this);
+        }
 
-            // "Row" children start new rows, and are themselves populated, other node types are additional colu
-            if (currentName == AZ::DocumentPropertyEditor::Nodes::Row::Name)
+        // clear any children before repopulating
+        if (!m_rowChildren.isEmpty() || !m_columnChildren.isEmpty())
+        {
+            if (notifyView && !m_rowChildren.isEmpty())
             {
-                DPEModelNode* newRow = AddChild(NodeType::RowNode, arrayIndex, false, QString::number(m_children.count()));
-                newRow->Populate(currentValue);
+                theModel->beginRemoveRows(myIndex, 0, GetRowChildCount() - 1);
+                ClearChildren();
+                theModel->endRemoveRows();
             }
-            else if (currentName == AZ::DocumentPropertyEditor::Nodes::Label::Name)
+            else
+            {
+                ClearChildren();
+            }
+        }
+
+        // populate type and string, if applicable
+        if (domName == AZ::DocumentPropertyEditor::Nodes::Adapter::Name)
+        {
+            m_type = NodeType::RootNode;
+        }
+        else if (domName == AZ::DocumentPropertyEditor::Nodes::Row::Name)
+        {
+            m_type = NodeType::RowNode;
+        }
+        else if (domName == AZ::DocumentPropertyEditor::Nodes::Label::Name)
+        {
+            m_type = NodeType::LabelNode;
+            m_displayString =
+                QString::fromUtf8(AZStd::string(domVal[AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString()).c_str());
+        }
+        else if (domName == AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Name)
+        {
+            m_type = NodeType::PropertyEditorNode;
+            auto& jsonBackend = GetModel()->GetBackend();
+            AZStd::string stringBuffer;
+            AZ::Dom::Utils::ValueToSerializedString(
+                jsonBackend, domVal[AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Value.GetName()], stringBuffer);
+            m_displayString = QString::fromUtf8(stringBuffer.c_str());
```

Comment:
nit: I think `fromUtf8` can take a size parameter as well

Reply:
I'm not clear on how this would help. Do you think this is faster or safer than relying on the null terminator?

Reply:
It would be faster in a minuscule manner than relying on a null-terminator, as `strlen` doesn't need to be performed over the c-string.

Reply:
may.. be? Depends on whether fromUtf8 is reading character by character or as memory block. But sure, happy to change it. Fixed now.